pkgs/by-name/li/libgphoto2/package.nix

```diff
@@ -1,20 +1,25 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, buildPackages
-, autoreconfHook
-, pkg-config
-, gettext
-, libusb1
-, libtool
-, libexif
-, libgphoto2
-, libjpeg
-, curl
-, libxml2
-, gd
+{
+  lib,
+  nixos,
+  stdenv,
+  fetchFromGitHub,
+  buildPackages,
+  autoreconfHook,
+  pkg-config,
+  gettext,
+  libusb1,
+  libtool,
+  libexif,
+  libgphoto2,
+  libjpeg,
+  curl,
+  libxml2,
+  gd,
 }:
-
+let
+  inherit (nixos { }) config;
+  group = config.programs.gphoto2.group or "camera";
```

Comment:
I don't see how this can work. Isn't this configuring a package build based on a static NixOS default value?

Reply:
If the user sets `programs.gphoto2.group = "foo"`, then the gphoto2 group will be "foo". If they don't set anything then it will be defaulted to "camera". Is that not what we want?

Reply:
The option already has a fallback but I thought we need this when where are not on a nixos system and just use nixpkgs. Not sure if we then eval the module system just for that.

Reply:
> If the user sets programs.gphoto2.group = "foo", then the gphoto2 group will be "foo".

Have you tested it?

Reply:
If you have some information, why not share it? I tried testing it, and first my entire computer crashed, and then there seems to be some unrelated issue with systemd on master.

Reply:
> If you have some information, why not share it?

I don't have any more information than is in this thread: I think this doesn't work because I think `nixos {} config` returns static defaults instead of taking into account the users config.

Reply:
Btw, you can probably test/apply this PR onto whatever nixpkgs revision your NixOS system is built from -- it doesn't have to be master.

Reply:
> I don't have any more information than is in this thread: I think this doesn't work because I think nixos {} config returns static defaults instead of taking into account the users config.

OIC, would absolutely be good to test this. 

> Btw, you can probably test/apply this PR onto whatever nixpkgs revision your NixOS system is built from -- it doesn't have to be master.

I'm afraid I don't know how to do that. The official [Nixpkgs/Reviewing changes](https://wiki.nixos.org/wiki/Nixpkgs/Reviewing_changes#Flakes) article doesn't seem to explain how to do that.

Reply:
Hmh, writing a test it turns out I don't know how to get the current `config.programs.gphoto2.group` value in `pkgs/by-name/li/libgphoto2/package.nix`. I've tried a few variants, but I don't understand how other packages are getting [things like `config.pulseaudio`](https://github.com/l0b0/nixpkgs/blob/gphoto2-group-name-config/pkgs/by-name/ae/aegisub/package.nix#L37-L37) seemingly without descending into `config.services` first.

Reply:
1. I think it's not possible. The only way would be to override the package inside a NixOS module.
2. config.pulseaudio is nixpkgs configuration, not NixOS.

Reply:
> I'm afraid I don't know how to do that.

Take the nixpkgs git working tree you already have (where this PR is made from), checkout a new branch from the nixpkgs revision your NixOS is built from, git cherry-pick the PR branch, run "NIX_PATH=nixpkgs=/path/to/my/nixpkgs nixos-rebuild ...".

Reply:
Thanks for the patience! I've changed this around to override the libgphoto2 group when enabling `programs.gphoto2`. What do you think?

Reply:
I think that will work.

However, I'm not so happy about having to rebuild software just to change group config. In your PR message you say the point is to not use magic string in config, but that can be solved by hardcoding the group in the package and making a read-only NixOS option that propagates that value.

Alternatively, if you really want to be able to change the group in NixOS config, I think the option docstring should warn about the rebuild cost.

Note: this is not me blocking the PR, but I'm not comfortable approving it either. If anyone else think it's fine, they can merge.

Reply:
Good points! I'll go for a read-only option instead. The main point was to be able to refer to the group in things like `users.users.<name>.extraGroups`, after all. How should I expose the group in libgphoto2? I can't seem to find any examples of arbitrary attributes like this in other packages.

Reply:
I've gone for `passthru.group` and a read-only option, so there should be no more rebuilds.

Reply:
[That doesn't work](https://github.com/NixOS/nixpkgs/actions/runs/14155119464/job/39653362389?pr=296832). Not sure how such a reference is meant to be done.

Reply:
About the CI error, try adding defaultText next to default. That should avoid the dependency it complains about.

Reply:
OK, let's see how this works.